File: src/main/java/com/github/zikani03/RepoInfo.java
package com.github.zikani03;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

/**
 * Subset of the Repository information provided by GitHub's API
 *
 * @date 2017-04-01
 */
@Data
@NoArgsConstructor
@AllArgsConstructor
@JsonIgnoreProperties(ignoreUnknown = true)
public class RepoInfo {
    @JsonProperty("name")
    private String name;

    @JsonProperty("full_name")
    private String fullName;

    @JsonProperty("size")
    private int size;

    @JsonProperty("language")
    private String language;

    @JsonProperty("git_url")
    private String gitUrl;
}

File: src/repo_info.rs
// This module represents repository information.

#[derive(Debug)]
pub struct RepoInfo {
    pub name: String,
    pub description: Option<String>,
}

impl RepoInfo {
    pub fn new(name: &str, description: Option<&str>) -> RepoInfo {
        RepoInfo {
            name: name.to_string(),
            description: description.map(|s| s.to_string()),
        }
    }
}